### PR TITLE
Switching to medallia headers

### PR DESCRIPTION
**Pull request by erdody**

### CODE REVIEW

File: linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala [hunk 1]
```diff
@@ -11,6 +11,28 @@ import java.net.URLEncoder
...
+  Mapping from linkerd header names to medallia header names.
+ */
+object MedalliaHeaders {
+  val medalliaPrefix = "X-Medallia"
+  val mapHeaderNames = collection.immutable.HashMap(
+    "l5d-sample" -> "X-Medallia-Tracing-Enabled",
+    "l5d-reqid" -> "X-Medallia-Tracer-Request-Id",
+    "l5d-ctx-trace" -> "X-Medallia-Tracer-Span-Id",
+    "L5d-sample" -> "X-Medallia-Tracing-Enabled",
+    "L5d-reqId" -> "X-Medallia-Tracer-Request-Id",
```
Comment (erdody) [inline linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala]: Can we do a case insensitive search so we don't duplicate these?

File: linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala [hunk 2]
```diff
@@ -114,31 +136,31 @@ object Headers {
      * deadline. Otherwise, outgoing requests MAY have a deadline.
      */
     object Deadline {
-      val Key = Prefix + "deadline"
+      val Key = MedalliaHeaders.getMedalliaHeaderName(Prefix + "deadline")
 
-      def read(v: String): FDeadline = {
+      def read(v: String): context.Deadline = {
```
Comment (erdody) [inline linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala]: Why was this changed? (and all the other FDeadline renames)

File: linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala [hunk 3]
```diff
@@ -11,6 +11,28 @@ import java.net.URLEncoder
...
 import scala.collection.breakOut
 
+/*
+  Mapping from linkerd header names to medallia header names.
+ */
+object MedalliaHeaders {
+  val medalliaPrefix = "X-Medallia"
+  val mapHeaderNames = collection.immutable.HashMap(
+    "l5d-sample" -> "X-Medallia-Tracing-Enabled",
+    "l5d-reqid" -> "X-Medallia-Tracer-Request-Id",
```
Comment (erdody) [inline linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala]: We already have an X-Medallia-Rpc-Request-Id, can that one replace this?

File: linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala [hunk 4]
```diff
@@ -11,6 +11,28 @@ import java.net.URLEncoder
 import java.util.Base64
```
Comment (erdody) [inline linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala:11]: Let's create an issue in upstream to add plugin support to change header names.

File: linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala [hunk 5]
```diff
@@ -11,6 +11,28 @@ import java.net.URLEncoder
...
 import scala.collection.breakOut
 
+/*
+  Mapping from linkerd header names to medallia header names.
+ */
+object MedalliaHeaders {
+  val medalliaPrefix = "X-Medallia"
```
Comment (erdody) [inline linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala]: The standard seems to be MedalliaPrefix (first letter uppercase) for object vals.

File: linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala (hunk 1 above)
Comment (AbhiMedallia) [reply linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala]: Sure. Done.

File: linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala [hunk 6]
```diff
@@ -11,6 +11,25 @@ import java.net.URLEncoder
...
+object MedalliaHeaders {
+  val MedalliaPrefix = "X-Medallia"
+  val mapHeaderNames = collection.immutable.HashMap(
+    "L5D-SAMPLE" -> "X-Medallia-Tracing-Enabled",
+    "L5D-REQID" -> "X-Medallia-Tracer-Request-Id",
+    "L5D-CTX-TRACE" -> "X-Medallia-Tracer-Span-Id"
+  )
+  def getMedalliaHeaderName(linkerdHeaderName: String): String = {
+    if (mapHeaderNames contains linkerdHeaderName.toUpperCase)
+      mapHeaderNames(linkerdHeaderName)
```
Comment (erdody) [inline linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala]: Don't we need to toUpperCase this line too?
Let's add a simple test to avoid regressions.
